src/mame/drivers/neogeo.cpp

```diff
@@ -2424,373 +2422,439 @@ CONS( 1990, aes,      0,       0,      aes,     aes,         aes_state,        e
+{
 	mv1_fixed(config);
-	NEOGEO_CONFIG_ONE_FIXED_CARTSLOT("boot_svcplus")
-MACHINE_CONFIG_END
+	cartslot_fixed(config, "boot_svcplus");
+}
 
-MACHINE_CONFIG_START(mvs_led_state::svcplusa)
+void mvs_led_state::svcplusa(machine_config &config)
+{
 	mv1_fixed(config);
-	NEOGEO_CONFIG_ONE_FIXED_CARTSLOT("boot_svcplusa")
-MACHINE_CONFIG_END
+	cartslot_fixed(config, "boot_svcplusa");
+}
 
-MACHINE_CONFIG_START(mvs_led_state::svcsplus)
+void mvs_led_state::svcsplus(machine_config &config)
+{
 	mv1_fixed(config);
-	NEOGEO_CONFIG_ONE_FIXED_CARTSLOT("boot_svcsplus")
-MACHINE_CONFIG_END
+	cartslot_fixed(config, "boot_svcsplus");
+}
 
-MACHINE_CONFIG_START(mvs_led_state::samsho5)
+void mvs_led_state::samsho5(machine_config &config)
+{
 	mv1_fixed(config);
-	NEOGEO_CONFIG_ONE_FIXED_CARTSLOT("k2k2_samsh5")
-MACHINE_CONFIG_END
+	cartslot_fixed(config, "k2k2_samsh5");
+}
 
-MACHINE_CONFIG_START(mvs_led_state::samsho5b)
+void mvs_led_state::samsho5b(machine_config &config)
+{
 	mv1_fixed(config);
-	NEOGEO_CONFIG_ONE_FIXED_CARTSLOT("boot_samsho5b")
-MACHINE_CONFIG_END
+	cartslot_fixed(config, "boot_samsho5b");
+}
 
-MACHINE_CONFIG_START(mvs_led_state::kof2003)
+void mvs_led_state::kof2003(machine_config &config)
+{
 	mv1_fixed(config);
-	NEOGEO_CONFIG_ONE_FIXED_CARTSLOT("pvc_kf2k3")
-MACHINE_CONFIG_END
+	cartslot_fixed(config, "pvc_kf2k3");
+}
 
-MACHINE_CONFIG_START(mvs_led_state::kof2003h)
+void mvs_led_state::kof2003h(machine_config &config)
+{
 	mv1_fixed(config);
-	NEOGEO_CONFIG_ONE_FIXED_CARTSLOT("pvc_kf2k3h")
-MACHINE_CONFIG_END
+	cartslot_fixed(config, "pvc_kf2k3h");
+}
 
-MACHINE_CONFIG_START(mvs_led_state::kf2k3bl)
+void mvs_led_state::kf2k3bl(machine_config &config)
+{
 	mv1_fixed(config);
-	NEOGEO_CONFIG_ONE_FIXED_CARTSLOT("boot_kf2k3bl")
-MACHINE_CONFIG_END
+	cartslot_fixed(config, "boot_kf2k3bl");
+}
 
-MACHINE_CONFIG_START(mvs_led_state::kf2k3pl)
+void mvs_led_state::kf2k3pl(machine_config &config)
+{
 	mv1_fixed(config);
-	NEOGEO_CONFIG_ONE_FIXED_CARTSLOT("boot_kf2k3pl")
-MACHINE_CONFIG_END
+	cartslot_fixed(config, "boot_kf2k3pl");
+}
 
-MACHINE_CONFIG_START(mvs_led_state::kf2k3upl)
+void mvs_led_state::kf2k3upl(machine_config &config)
+{
 	mv1_fixed(config);
-	NEOGEO_CONFIG_ONE_FIXED_CARTSLOT("boot_kf2k3upl")
-MACHINE_CONFIG_END
+	cartslot_fixed(config, "boot_kf2k3upl");
+}
 
-MACHINE_CONFIG_START(mvs_led_state::samsh5sp)
+void mvs_led_state::samsh5sp(machine_config &config)
+{
 	mv1_fixed(config);
-	NEOGEO_CONFIG_ONE_FIXED_CARTSLOT("k2k2_sams5s")
-MACHINE_CONFIG_END
+	cartslot_fixed(config, "k2k2_sams5s");
+}
 
-MACHINE_CONFIG_START(mvs_led_state::neogeo_mj)
+void mvs_led_state::neogeo_mj(machine_config &config)
+{
 	mv1_fixed(config);
 	set_default_bios_tag("japan");
 
 	//no joystick panel
-	MCFG_DEVICE_REMOVE("edge")
-	MCFG_NEOGEO_CONTROL_EDGE_CONNECTOR_ADD("edge", neogeo_arc_edge, "", false)
+	config.device_remove("edge");
+	NEOGEO_CTRL_EDGE_CONNECTOR(config, m_edge, neogeo_arc_edge, "", false);
```

Comment:
Since you're not using macros any more, you can get rid of the `device_remove` and use `config.replace()` in the following line to replace the device in a single step.  Same applies for the controller ports below.